internal/api/v1/application/deploy.go

```diff
@@ -64,14 +51,30 @@ func (hc Controller) Deploy(c *gin.Context) apierror.APIErrors {
 		return apierror.InternalError(err, "failed to get access to a kube client")
 	}
 
+	return DeployCurrentAppState(c, log, ctx, cluster, req.App, username, req.ImageURL, req.Stage.ID, &req.Origin)
+}
+
+func DeployCurrentAppState(c *gin.Context, log logr.Logger, ctx context.Context, cluster *kubernetes.Cluster, app models.AppRef, username, imageURL, stageID string, origin *models.ApplicationOrigin) apierror.APIErrors {
```

Comment:
We probably don't need the gin.Context, and the logger can/should be fetched from the `context.Context`:
```golang
log := requestctx.Logger(ctx)
```

Reply:
Agree on `log` ... gin is currently used for the OKReturn at the end ... However that likely has to be moved out as it looks to be specific to the deploy endpoint and should not be used for the updating endpoints.

Reply:
It was quick work yesterday between meetings, not too much thought went into the signature then.

Reply:
Commit d6059482226188462b3cd17b7fba44e5004c035a